classes/ActionScheduler_wpPostStore.php

```diff
@@ -119,12 +119,64 @@ protected function make_action_from_post( $post ) {
 		}
 		$group = wp_get_object_terms( $post->ID, self::GROUP_TAXONOMY, array('fields' => 'names') );
 		$group = empty( $group ) ? '' : reset($group);
-		if ( $post->post_status == 'pending' ) {
-			$action = new ActionScheduler_Action( $hook, $args, $schedule, $group );
-		} else {
-			$action = new ActionScheduler_FinishedAction( $hook, $args, $schedule, $group );
+
+		return ActionScheduler::factory()->get_stored_action( $this->get_action_status_by_post_status( $post->post_status ), $hook, $args, $schedule, $group );
+	}
+
+	/**
+	 * @param string $post_status
+	 *
+	 * @return string
+	 */
+	protected function get_action_status_by_post_status( $post_status ) {
+
+		switch ( $post_status ) {
+			case 'any' :
+				$post_status = 'all';
+				break;
+			case 'publish' :
+				$action_status = self::STATUS_COMPLETE;
+				break;
+			case 'trash' :
+				$action_status = self::STATUS_CANCELED;
+				break;
+			default :
+				if ( ! array_key_exists( $post_status, $this->get_status_labels() ) ) {
+					throw new InvalidArgumentException( sprintf( 'Invalid post status: "%s". No matching action status available.', $post_status ) );
+				}
+				$action_status = $post_status;
+				break;
 		}
-		return $action;
+
+		return $action_status;
+	}
+
+	/**
+	 * @param string $action_status
+	 *
+	 * @return string
```

Comment:
This should include an `@throws` tag indicating what causes the `InvalidArgumentException` to be thrown.